### PR TITLE
Improve moderator's emphasizing chat messages.

### LINKED ISSUE #14243: Improve moderator chat messages emphasizing (live session and recording)
We recently implemented a way to emphasize moderator messages.

We currently have two limitations in this feature:

1 - If a user is demoted, the messages sent while user was a moderator are de-emphasized.
2 - The recordings are not emphasizing the messages.

**Describe the solution you'd like**
As proposed by @GuiLeme [here](https://github.com/bigbluebutton/bigbluebutton/issues/13547#issuecomment-1026771918), akka will store "per message" the current role of the user (when the message is sent).

This information will be propagated to html5-server and to recording archive (`events.xml`).

**Describe alternatives you've considered**
We considered calculating the "time periods" that the user was moderator, to emphasize message accordingly, however it added complexity to the code and didn't solved the issue with the UI during the meeting.

**Additional context**

This issue is a follow up of https://github.com/bigbluebutton/bigbluebutton/issues/13547.
This issue also makes PR https://github.com/bigbluebutton/bigbluebutton/pull/14227 obsolete.

### PR DESCRIPTION
**Pull request by GuiLeme**
### What does this PR do?

It does 2 things in total: 
 1. It marks the message from the chat with a property called role, so that we don't use the updated user's role, instead, we use  the role on which the user sent the message.
 2. It updates the recording chat to keep track of the role changes.

The first point helps us with both keeping track of what role was the user when the message was sent,  and not re-rendering the chat on every change of users' role.
The second point is important so that we can resolve issue #13547 

Closes #14243

Edit: I already made a PR to the `bbb-playback` repository, so that with this PR and that one (https://github.com/bigbluebutton/bbb-playback/pull/170), the new feature described by #14243 is complete

### CODE REVIEW

**Comment by antobinary** [comment]
Nice work :confetti_ball: :smiley_cat:

File: akka-bbb-apps/src/main/scala/org/bigbluebutton/endpoint/redis/RedisRecorderActor.scala [hunk 1]
```diff
@@ -139,6 +139,8 @@ class RedisRecorderActor(
       ev.setMeetingId(msg.header.meetingId)
       ev.setSenderId(msg.body.msg.sender.id)
       ev.setMessage(msg.body.msg.message)
+      ev.setSenderRole(msg.body.msg.sender.role)
+      ev.setChatEmphasizedText(msg.body.msg.chatEmphasizedText)
```
Comment (pedrobmarin) [inline akka-bbb-apps/src/main/scala/org/bigbluebutton/endpoint/redis/RedisRecorderActor.scala:143]:
```suggestion
      ev.setSenderRole(msg.body.msg.sender.role)
      
      val isModerator = msg.body.sender.role == "MODERATOR"
      ev.setChatEmphasizedText(msg.body.msg.chatEmphasizedText && isModerator)
```

File: akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/groupchats/SendGroupChatMessageMsgHdlr.scala [hunk 2]
```diff
@@ -15,11 +15,13 @@ trait SendGroupChatMessageMsgHdlr extends HandlerHelpers {
              liveMeeting: LiveMeeting, bus: MessageBus): MeetingState2x = {
 
     var chatLocked: Boolean = false
+    var userRole: String = Roles.VIEWER_ROLE
 
     for {
       user <- Users2x.findWithIntId(liveMeeting.users2x, msg.header.userId)
       groupChat <- state.groupChats.find(msg.body.chatId)
     } yield {
+      userRole = user.role
```
Comment (pedrobmarin) [inline akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/groupchats/SendGroupChatMessageMsgHdlr.scala:24]: Looks like this isn't used anywhere

**Review by pedrobmarin — CHANGES_REQUESTED** [review]

File: akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/groupchats/SendGroupChatMessageMsgHdlr.scala (hunk 2 above)
Comment (GuiLeme) [reply akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/groupchats/SendGroupChatMessageMsgHdlr.scala:24]: Definetly

File: bbb-common-message/src/main/scala/org/bigbluebutton/common2/msgs/GroupChatMsg.scala [hunk 3]
```diff
@@ -5,9 +5,9 @@ object GroupChatAccess {
   val PRIVATE = "PRIVATE_ACCESS"
 }
 
-case class GroupChatUser(id: String, name: String = "")
-case class GroupChatMsgFromUser(correlationId: String, sender: GroupChatUser, message: String)
-case class GroupChatMsgToUser(id: String, timestamp: Long, correlationId: String, sender: GroupChatUser, message: String)
+case class GroupChatUser(id: String, name: String = "", role: String = "")
```
[the remark this answers is not in the document]
Comment (GuiLeme) [reply bbb-common-message/src/main/scala/org/bigbluebutton/common2/msgs/GroupChatMsg.scala:8]: Also, done.

File: akka-bbb-apps/src/main/scala/org/bigbluebutton/endpoint/redis/RedisRecorderActor.scala [hunk 4]
```diff
@@ -141,6 +141,10 @@ class RedisRecorderActor(
       ev.setMeetingId(msg.header.meetingId)
       ev.setSenderId(msg.body.msg.sender.id)
       ev.setMessage(msg.body.msg.message)
+      ev.setSenderRole(msg.body.msg.sender.role)
+      
+      val isModerator = msg.body.sender.role == "MODERATOR"
```
Comment (pedrobmarin) [inline akka-bbb-apps/src/main/scala/org/bigbluebutton/endpoint/redis/RedisRecorderActor.scala:146]:
```suggestion
      val isModerator = msg.body.msg.sender.role == "MODERATOR"
```
I was the one who suggested this but I didn't notice it was missing a `msg` after the `body` prop. My bad!

**Review by pedrobmarin — APPROVED** [review]

**Pull request merged**